test/e2e/ginkgo_operator-sdk.go

```diff
@@ -0,0 +1,230 @@
+	ginkgoSeed, err := getGinkoSeed()
+	if err != nil {
+		return err
+	}
+
+	globalManFile := fmt.Sprintf("%s/global-manifest-%s.yaml", os.TempDir(), ginkgoSeed)
+	err = os.Remove(globalManFile)
+	if err != nil {
+		return err
+	}
+
+	namespacedManFile := fmt.Sprintf("%s/namespace-manifest-%s.yaml", os.TempDir(), ginkgoSeed)
+	err = os.Remove(namespacedManFile)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getArg(argName string) (bool, string) {
+	for i, arg := range os.Args {
+		if arg == argName {
+			return true, os.Args[i+1]
+		} else if strings.HasPrefix(arg, argName+"=") {
+			argAndValue := strings.SplitN(arg, "=", 2)
+			return true, argAndValue[1]
+		}
+	}
+	return false, ""
+}
+
+func getGinkgoNode() int {
+	defined, ginkgoNodeString := getArg("--ginkgo.parallel.node")
+	if !defined {
+		return 1
+	}
+	ginkgoNode, err := strconv.Atoi(ginkgoNodeString)
+	if err != nil {
+		fmt.Printf("Error: %s", err.Error())
+		return 0
+	}
+	return ginkgoNode
+}
+
+func getGinkoSeed() (string, error) {
+	defined, ginkgoSeed := getArg("--ginkgo.seed")
+	if !defined {
+		return "", errors.New("unable to find --ginko.seed argument")
+	}
+	return ginkgoSeed, nil
+}
+
+func getRootDir() (string, error) {
+	rootDir, err := os.Getwd()
+	if err != nil {
+		return "", err
+	}
+
+	for info, err := os.Stat(rootDir + "/LICENSE"); os.IsNotExist(err) || (err == nil && info.IsDir()); {
+		rootDir = filepath.Dir(rootDir)
+		if rootDir == "/" {
+			return "", errors.New("Failed to determine root directory")
+		}
+		info, err = os.Stat(rootDir + "/LICENSE")
+	}
+
+	return rootDir, nil
+}
+
+func isGinkgoRunner() bool {
+	defined, _ := getArg("--ginkgo.seed")
+	return defined
+}
+
+func populateFrameworkScheme() error {
+	err := framework.AddToFrameworkScheme(apis.AddToScheme, &operator.BuildList{})
+	if err != nil {
+		Logf("Failed to add BuildList to schema %s", err.Error())
+		return err
+	}
+
+	err = framework.AddToFrameworkScheme(apis.AddToScheme, &operator.BuildStrategyList{})
+	if err != nil {
+		Logf("Failed to add BuildStrategyList to schema %s", err.Error())
+		return err
+	}
+
+	err = framework.AddToFrameworkScheme(apis.AddToScheme, &operator.ClusterBuildStrategyList{})
+	if err != nil {
+		Logf("Failed to add ClusterBuildStrategyList to schema %s", err.Error())
+		return err
+	}
+
+	err = framework.AddToFrameworkScheme(pipelinev1.AddToScheme, &pipelinev1.TaskList{})
+	if err != nil {
+		Logf("Failed to add TaskList to schema %s", err.Error())
+		return err
+	}
+
+	err = framework.AddToFrameworkScheme(pipelinev1.AddToScheme, &pipelinev1.TaskRunList{})
+	if err != nil {
+		Logf("Failed to add TaskRunList to schema %s", err.Error())
+		return err
+	}
+
+	return nil
+}
+
+func startLocalOperator() (*exec.Cmd, error) {
+	rootDir, err := getRootDir()
+	if err != nil {
+		return nil, err
+	}
+
+	// define the operator command
+	operatorCmd := exec.Command("operator-sdk", "run", "--local", "--verbose", "--namespace", os.Getenv("TEST_NAMESPACE"))
```

Comment:
I feel a bit uneasy on running the `operator-sdk` command inside out. Could we have regular the regular approach, where `operator-sdk` drives the test workflow instead?

Reply:
From the root directory, you can still run `operator-sdk test local ./test/e2e --up-local --namespace build-examples --verbose --go-test-flags "-timeout=20m"`. This still works even with my changes.

But, using `operator-sdk test` it is NOT possible to use the Ginkgo CLI directly. `operator-sdk test` always invokes `go test`. See https://github.com/operator-framework/operator-sdk/blob/v0.15.x/cmd/operator-sdk/test/local.go#L228 and https://github.com/operator-framework/operator-sdk/blob/v0.15.x/internal/util/projutil/exec.go#L78. Through go test, it is possible to run Ginkgo tests, but one requires the Ginkgo CLI to use features like `nodes` (for parallelism) or `focus` and `skip` (to only run selective tests).

Reply:
After the discussion I had with @qu1queee last week, I removed the option to start the test through `operator-sdk test` as this simplifies the code.

With the Ginkgo CLI being the main and only entry point, I also envision further enhancements related to the management of the operator as part of the test easier. We specifically have the requirement to use an existing operator in our environment and we also cannot check its availability because the user does not have access to the deployment. We just have to assume it is there and run the test. Such cases can be easily handled in the before suite code while the `operator-sdk test` command line brings in expectations that we cannot always fulfill. But other options that we could offer include to deploy the operator as a container image through `kubectl apply` or `helm` - which is more realistic than running it locally (while running it locally is a really nice way while doing development on it).